test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -53,6 +62,50 @@
 public abstract class MapperTestCase extends MapperServiceTestCase {
     protected abstract void minimalMapping(XContentBuilder b) throws IOException;
 
+    protected void field(XContentBuilder builder) throws IOException {
+        builder.field("field");
+        fieldValue(builder);
+    }
+
+    protected abstract void fieldValue(XContentBuilder builder) throws IOException;
+
+    public final void testExistsQuery() throws IOException {
+        MapperService mapperService = createMapperService(fieldMapping(this::minimalMapping));
+        ParsedDocument doc = mapperService.documentMapper().parse(source(this::field));
+        withLuceneIndex(mapperService, iw -> iw.addDocument(doc.rootDoc()), reader -> {
+            QueryShardContext queryShardContext = createQueryShardContext(mapperService);
+            MappedFieldType fieldType = mapperService.fieldType("field");
+            assertExistsQuery(fieldType, queryShardContext, reader);
+        });
+        assertParseMinimalWarnings();
+    }
+
+    protected void assertExistsQuery(MappedFieldType fieldType, QueryShardContext queryShardContext, IndexReader reader) {
```

Comment:
I think you can change this to just take `MapperService`.  Then we have the default `testExistsQuery` impl which calls this with a minimal mapping; and mappers which have norms or doc_values settings can then have their own tests that call this with the relevant configuration.

Reply:
this is implemented now, thanks for the tip.